packages/react-native/android/src/main/java/com/dashx/sdk/DashXClient.kt

```diff
@@ -1,37 +1,139 @@
 package com.dashx.sdk
 
 import android.content.SharedPreferences
-import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.*
+import com.google.gson.Gson
+import okhttp3.*
+import okhttp3.Callback
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.io.IOException
 import java.util.*
 
 
 class DashXClient {
-    private var reactApplicationContext: ReactApplicationContext? = null
-    private val anonymousUid: String? = null
+    private val tag = DashXClient::class.java.simpleName
+    private var anonymousUid: String? = null
+    private var baseURI: String = "https://api.dashx.com/v1"
+    private var publicKey: String? = null
+    private var uid: String? = null
 
-    fun setReactApplicationContext(reactApplicationContext: ReactApplicationContext?) {
-        this.reactApplicationContext = reactApplicationContext
+    private val httpClient = OkHttpClient()
+    private val gson = Gson()
+    private val json = "application/json; charset=utf-8".toMediaType()
+
+    var reactApplicationContext: ReactApplicationContext? = null
+
+    fun setBaseURI(baseURI: String) {
+        this.baseURI = baseURI
     }
 
-    fun getReactApplicationContext(): ReactApplicationContext? {
-        return reactApplicationContext
+    fun setPublicKey(publicKey: String) {
+        this.publicKey = publicKey
     }
 
     fun generateAnonymousUid() {
         val dashXSharedPreferences: SharedPreferences = getDashXSharedPreferences(reactApplicationContext!!.applicationContext)
-        var anonymousUid = dashXSharedPreferences.getString(SHARED_PREFERENCES_KEY_ANONYMOUS_UID, null)
+        val anonymousUid = dashXSharedPreferences.getString(SHARED_PREFERENCES_KEY_ANONYMOUS_UID, null)
         if (anonymousUid != null) {
-            anonymousUid = this.anonymousUid
+            this.anonymousUid = anonymousUid
         } else {
-            anonymousUid = UUID.randomUUID().toString()
+            this.anonymousUid = UUID.randomUUID().toString()
             dashXSharedPreferences.edit()
-                .putString(SHARED_PREFERENCES_KEY_ANONYMOUS_UID, anonymousUid)
+                .putString(SHARED_PREFERENCES_KEY_ANONYMOUS_UID, this.anonymousUid)
                 .apply()
         }
     }
 
+    fun identify(uid: String?, options: ReadableMap?, promise: Promise) {
```

Comment:
For regular users, the call looks like this:

```
identify('123')
```

This simply sets `uid` in DashXClient.

For anonymous users, the call looks like this:

```
identify({
  firstName: 'John'
})
```

This makes the API call with `anonymous_uid`.